### PR TITLE
DAOS-14559 ci: Create Actions jobs for isort linting and version checking.

### PR DESCRIPTION
**Pull request by ashleypittman**
Create a new generic linting job and add isort checking.
Pin versions of python checking tools.
Add job to master that checks versions are up-to-date.

Fix isort options for line-length and blacklisting.

Resolve isort and codespell issues.

Overall this increases coverage with new checks, gives us a common
linting job which should be easier to manage that multiple ones and
allows us to run PRs with pinned versions to prevent upstream changes
blocking development but still fails master checking when updates are
available.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by jolivier23 — APPROVED** [review]

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: actions/setup-python@v3
+      - uses: isort/isort-action@master
+        with:
+          requirementsFiles: "requirements.txt utils/cq/requirements.txt"
+      - name: Run on SConstruct file.
+        run: isort --check-only SConstruct
+      - name: Run on build files.
+        run: find . -name SConscript | xargs isort
```
Comment (daltonbohning) [inline .github/workflows/linting.yml:31]: Maybe this is what the comment above was referring to, but do we need `--check-only` for the SConscript run?

File: utils/cq/requirements.txt [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 flake8<6.0.0
-isort
-paramiko
-pyenchant
-pylint
-yamllint
+isort==5.12
+pylint==3.0.2
+yamllint==1.32.0
+codespell==2.2.6
```
Comment (daltonbohning) [inline utils/cq/requirements.txt:15]: Just pointing out this requires a newer python. My distro default was python3.6, which did not have isort 5.12 available. After upgrading to python3.9, this works. I talked with @ashleypittman and really we need better dev setup documentation

**Review by daltonbohning — APPROVED** [review]

**Review by daltonbohning — CHANGES_REQUESTED** [review]

File: .github/workflows/linting.yml (hunk 1 above)
Comment (ashleypittman) [reply .github/workflows/linting.yml]: Good spot, repushed.

File: utils/cq/requirements.txt (hunk 2 above)
Comment (ashleypittman) [reply utils/cq/requirements.txt:15]: I agree, but it's no reason for CI and power users to have access to the latest versions of the tools.

**Review by jolivier23 — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**